src/components/common/Select.js

```diff
@@ -1,43 +1,107 @@
-import React from 'react';
+import React, { useState, useRef } from 'react';
 import styled, { css } from 'styled-components';
 import PropTypes from 'prop-types';
 
+import useOutsideAlerter from 'hooks/useOutsideAlerter';
 import DropdownArrow from 'components/common/DropdownArrow';
 
-const StyledSelect = styled.select`
+const StyledSelect = styled.div`
```

Comment:
Never used styled components before so I don't know what's the standard way of using them. Is there a way to have the styling defined in another file?  I feel like the styles are getting in the way and making the logic harder to read.

Reply:
does it really? The actual render of the Select is 20 lines. And there is not really a standard for that, you can do it if you want, but it would make more sense to do it, if you were to reuse the components across the app.

Reply:
I found this article https://tech.decisiv.com/structuring-our-styled-components-part-i-2bf21fa64b28

Reply:
It talks of using styled components borrowing ideas from BEM, if you can take a look and think it will make it easier to understand, I can follow it and have like a file for the `Select`, and a file for the `Select.Option`

Reply:
Yeah, maybe is too much, I wouldn't like to follow 100% what the post says, but at least having two files, one for `Select`, and one for the `SelectOption`. Maybe I can iterate first step like this

Reply:
Yep, other idea is to have a file with your JSX markdown (so the plain component as if you were not using styled) and then on your index.js wrap that with your styles and re-export it. That could work too, just a suggestion though

Reply:
Agree with @aguscha333 here, sometimes is difficult for me to read the components with all the css in the way

Reply:
Agreed, the reading of components with styled components in the middle is kinda hard for me too. I believe this is one of the first projects that uses styled components in RS, so we should try to see how it can fit our needs. I've checked slack and @bernabe9 seems to have some experience using them, so we could benefit from his advice.

Reply:
@manuelcheveste I checked the link you sent. I have now a better understanding of what styled components are. I am still not 100% sold on styled components.
I realize that putting the styles in another file might be hard with styled components. If you know a way to do it I would love to see an example of that implementation to see if it's worth considering.
Aside from that I think that there are some improvements that can be done to make this file more readable. I would structure the blocks as the post did. Have a `Select` folder with the block code in `index.js` and have a separate file for each element. From what I saw elements are mostly just their styling so i don't mind mixing styling with element file.

Reply:
@aguscha333 Do you know redux ducks? Is about putting related redux logic in one file, e.g. having user actions, action creators, reducer and thunks. Is so simple to understand what a reducer does using this, and you don't have to keep track of 5 files just for one reducer.

I really encourage doing this, that's why I love using styled components, what's the point of splitting components and css into different files, if they are so closely related.

So I don't complain if it's hard to split styles, it is just about like going against the gist of styled components.

Reply:
It is my fault ofc, that this file is now too complicated to read, maybe, `Select` and `SelectOptions` are not that closely related, so I will split them, and I hope that gets easier for all to read.

Reply:
Yeah after reading the post I got that putting styles in a different file was probably not how styled components were intended to be used. Just wanted to know if it was a possible alternative that didn't go to much against styled components guidelines.

The point is separating logic form just simple styling. IMO it makes it easier to read. I guess styled components are just not my cup of tea and neither is redux ducks.

That being said, I still think that you should split the file into a folder with  a block component file and its elements in separate files to reduce the complexity of this file.

Reply:
Can we open an issue about doing this block/element stuff? Because if I do it now, I should also do it with every other component in the app for consistency. This will take time, and I would prefer focusing on my sprint todos. So the least I can do now is to split select and options

Reply:
Also in this issue, I could talk with Bernabe as @missx said, and decide which is the best way to handle large files when using styled components.

Reply:
👍

Reply:
👍 Sounds good, I think this kind of stuff works better when discussed IRL.

Reply:
Thanks! Sure, we can set up a meeting and everyone who is interested in it can and is welcome to join

Reply:
https://github.com/rootstrap/open-form-admin-webapp/issues/6
There's the issue, thanks all for the suggestions.